### ISSUE TITLE
RPC Argument Evolution

### ISSUE THREAD

**Issue by norgor** [capnproto/capnproto#1808]
I was wondering how the evolution of RPC arguments can be done, as I could not find any documentation on it. Specifically, can method arguments be changed to a struct (or the reverse), while ensuring backwards compatibility?

E.g. changing
```capnp
interface Greeter @0xf702ad8db05c5167 {
    greet @0 (name :Text) -> (greeting :Text);
}
```
to
```capnp
struct GreetArgs {
    name @0 :Text;
}
struct Greeting {
    greeting @0 :Text;
}
interface Greeter @0xf702ad8db05c5167 {
    greet @0 GreetArgs -> Greeting;
}
```

**Comment by kentonv** [comment]
Yes. A parenthesized parameter or result list is just an anonymous struct -- is encoded exactly the same as a struct containing the each parameter as a field, numbered in the order they appear in the list. You can swap this out for an actual struct type as shown in your example.

I guess the docs don't actually cover the syntax to specify a single struct type as the params or results. I'll leave this issue open for documenting that, and the compatibility properties of it.

**Comment by norgor** [comment]
Thanks a lot for the quick response! 

If you feel this should be in the documentation, I would be glad to contribute. Otherwise, I think this issue can be closed.
